File: src/main/java/com/github/nsc/de/shake/parser/node/logical/LogicalSmallerEqualsNode.java
package com.github.nsc.de.shake.parser.node.logical;

import com.github.nsc.de.shake.parser.node.ValuedNode;
import com.github.nsc.de.shake.util.characterinput.position.PositionMap;

public class LogicalSmallerEqualsNode extends LogicalCompareNode {

    public LogicalSmallerEqualsNode(PositionMap map, ValuedNode left, ValuedNode right) { super(map, left, right); }

    @Override
    public String getOperator() {
        return "<=";
    }

}

File: parser/src/commonMain/kotlin/com/github/shakelang/shake/parser/node/logical/LogicalSmallerEqualsNode.kt
package com.github.shakelang.shake.parser.node.logical

import com.github.shakelang.shake.parser.node.ValuedNode
import com.github.shakelang.shake.util.characterinput.position.PositionMap

class LogicalSmallerEqualsNode(map: PositionMap, left: ValuedNode, right: ValuedNode) :
    LogicalCompareNode(map, left, right) {
    override val operator: String
        get() = "<="

    override fun toJson(): Map<String, *> =
        mapOf("name" to "LogicalSmallerEqualsNode", "left" to left, "right" to right)
}